src/compiler/sys.ts

```diff
@@ -876,6 +877,15 @@ namespace ts {
                 bufferFrom,
                 base64decode: input => bufferFrom(input, "base64").toString("utf8"),
                 base64encode: input => bufferFrom(input).toString("base64"),
+                require: (baseDir, moduleName) => {
+                    try {

```

Comment:
Should this `try` wrap `resolveJSModule`? I'm not sure that should be allowed to fail like `require`.

Reply:
That's what we're already doing in src/tsserver/server.ts. In any event, the error is still available.

Reply:
🤷‍♂ It'll just be odd if, at some point in the future, it fails with a `Cannot access field 'kind' on type 'undefined'` error in our own code or something and it's just swallowed.

Reply:
It's not swallowed, per se. The error is returned in the result. We generally try to avoid catching/throwing exceptions in our code if possible (such as readFile returning `string | undefined`).

Reply:
Right, we wanna avoid catch/throwing fs and other kinds of node errors, but implementation problems and debug assertions we'd ideally like to still throw, loudly. That's why I was wondering if the scope of the try/catch was too broad.

Reply:
In cases like tsserver and the plugin API prototype, we inspect the `error` property and report it either in the server log or via a diagnostic, so the error itself is not lost. I'm explicitly *not* using the error property for the extended debug functionality.